### ISSUE TITLE
mutmut doesn't allow non-unicode characters

### ISSUE THREAD

**Issue by carloscaicedos** [boxed/mutmut#331]
Hi.
mumut is raising a UnicodeDecodeError when a file contains non-unicode characters, such as "ú".

```
1. Using cached time for baseline tests, to run baseline again delete the cache file

2. Checking mutants
⠸ 5/9  🎉 1  ⏰ 0  🤔 0  🙁 4  🔇 0Process check_mutants:
Traceback (most recent call last):
  File "C:\Program Files\Python312\Lib\multiprocessing\process.py", line 314, in _bootstrap
    self.run()
  File "C:\Program Files\Python312\Lib\multiprocessing\process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "D:\Repos\Python\mutation-testing\.venv\Lib\site-packages\mutmut\__init__.py", line 749, in check_mutants
    status = run_mutation(context, feedback)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Repos\Python\mutation-testing\.venv\Lib\site-packages\mutmut\__init__.py", line 794, in run_mutation
    survived = tests_pass(config=config, callback=callback)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Repos\Python\mutation-testing\.venv\Lib\site-packages\mutmut\__init__.py", line 865, in tests_pass
    returncode = popen_streaming_output(config.test_command, callback, timeout=config.baseline_time_elapsed * 10)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Repos\Python\mutation-testing\.venv\Lib\site-packages\mutmut\__init__.py", line 1051, in popen_streaming_output
    line = line.decode("utf-8")
           ^^^^^^^^^^^^^^^^^^^^
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xed in position 51: invalid continuation byte
```

```
mi_calc
├── setup.cfg
├── src
│   └── calculadora.py
└── tests
    └── test_calculadora.py
```

```
#calculadora.py

def sumar(a: float, b: float) -> float:
    if a < 0 or b < 0:
        print("No se pueden sumar números negativos.")
        raise ValueError("No se pueden sumar números negativos.")

    return a + b


def dividir(a: float, b: float) -> float:
    if b == 0:
        print("No se puede dividir por cero.")
        raise ValueError("No se puede dividir por cero.")

    return a / b

```

**Comment by boxed** [comment]
Your python source code should always be in utf8. It seems this file is not which is incorrect.

**Comment by carloscaicedos** [comment]
All my code is UTF-8; Visual Studio Code saves all my files in UTF-8. When I test mutmut without the letter ‘ú’, it works fine, but when I include the ‘ú’, mutmut crashes. It’s worth mentioning that this issue is not related to my code or my files.

**Comment by boxed** [comment]
Could you upload the file?

**Comment by carloscaicedos** [comment]
These are the files.

[project.zip](https://github.com/user-attachments/files/16405029/project.zip)

**Comment by boxed** [comment]
I downloaded that file, extracted it and mutmut runs cleanly through all files.

From what I can tell there is no file with `ú` in it. If I ADD that character randomly to a string inside `calculatora.py` it works fine.

I'm going to repeat myself: your file is not utf-8 encoded. Your belief that it is, is incorrect. The computer is clearly telling you that it's incorrect, and you need to trust the computer.

Also, if you want help, you need to send a file that reproduces the problem. You did not do this, you sent a file WITHOUT the problem.

**Comment by carloscaicedos** [comment]
My bad, I uploaded the wrong files.

Here are the correct files, plus a video that shows the error, and after removing the ‘ú,’ it works fine.

[mutation-testing.zip](https://github.com/user-attachments/files/16405118/mutation-testing.zip)

https://github.com/user-attachments/assets/20e4ac00-0fe9-4ece-9dc8-8d01ce8301df

**Comment by boxed** [comment]
You AGAIN  uploaded the file with just a plain u.

**Comment by carloscaicedos** [comment]
I apologize for the error; these are the files with ‘ú’.

[mutation-testing.zip](https://github.com/user-attachments/files/16405129/mutation-testing.zip)

Could you see the video?

**Comment by boxed** [comment]
Still doesn't reproduce. Anyway, you have to figure out what the issue is I think. Clearly you can't produce a file that is broken.

**Comment by carloscaicedos** [comment]
Hi. Could you see the video?

**Comment by carloscaicedos** [comment]
I started the test from scratch and noticed that the exception only occurs on Windows. I tested it on Linux using WSL, and mutmut works there. However, when I run the test on Windows, the exception occurs when mutating text strings that contain characters like á, é, í, ó, ú. 

I conducted the tests on two different PCs—one with Windows 11 and the other with Windows 10—and the exception occurs in both cases.
